src/focus.service.ts

```diff
@@ -421,10 +380,17 @@ export class FocusService {
 
   public createArcEvent(direction: Direction): ArcEvent {
     const directional = isDirectional(direction);
+    let element: HTMLElement;
+    if (directional) {
+      element = this.findNextFocusByRaycast(direction);
+      if (!element) {
+        element = this.findNextFocus(direction);
```

Comment:
That's right. It's a heuristic tradeoff that is optimized for the 95% of the common cases.